core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -7880,7 +7882,223 @@ components:
+            * Mint resources for previewing trades with resources you don't own.
+              If doing this, be warned: only resources which were potentially mintable/burnable
+              at creation time will be mintable/burnable, due to feature flags on the resource.
+            
+            Warning: this mode of operation is quite a departure from normal operation:
+            * Calculated fees will likely be lower than a standard execution.
+            * This mode can subtly break invariants some dApp code might rely on, or result in unexpected
+              behaviour, so the execution result might not be valid for your needs. For example,
+              if this flag was used to mint pool units to preview a redemption (or some dApp interaction which
+              behind the scenes redeemed them), they'd redeem for less than they're currently worth,
+              because the blueprint code relies on the total supply of the pool units to calculate their
+              redemption worth, and you've just inflated the total supply through the mint operation.
+          type: boolean
+####################################
+# REQUEST: /transaction/preview-v2 #
+####################################
+    TransactionPreviewV2Request:
+      type: object
+      required:
+        - network
+        - preview_transaction
+      properties:
+        network:
+          $ref: "#/components/schemas/NetworkIdentifier"
+        at_ledger_state:
+          $ref: "#/components/schemas/LedgerStateSelector"
+        preview_transaction:
+          $ref: "#/components/schemas/PreviewTransaction"
+        flags:
+          $ref: "#/components/schemas/PreviewFlags"
+        options:
+          $ref: "#/components/schemas/TransactionPreviewV2ResponseOptions"
+    TransactionPreviewV2ResponseOptions:
+      type: object
+      description: A set of flags to configure the response of the transaction preview.
+      properties:
+        core_api_receipt:
+          type: boolean
+          description: | 
+            This flag controls whether the preview response will include a Core API receipt or not.
+            If not provided, this defaults to `false` and no core api receipt is provided in
+            the response.
+        radix_engine_toolkit_receipt:
+          type: boolean
+          description: | 
+            This flag controls whether the preview response will include a Radix Engine Toolkit serializable
+            receipt or not. If not provided, this defaults to `false` and no toolkit receipt is provided in
+            the response.
+        logs:
+          type: boolean
+          description: | 
+            This flag controls whether the preview response will include execution logs.
+            If not provided, this defaults to `false` and no logs will be provided in the response.
+    PreviewTransactionType:
+      type: string
+      enum:
+        - Compiled
+    PreviewTransaction:
```

Comment:
as this is used only in v2 maybe we should name It PreviewTransactionV2 ?

Reply:
Well it's intended not just to be V2 transactions. e.g. it could include PreviewTransactionV3